### PR TITLE
Add some cases for virsh metadata

### PR DESCRIPTION
**Pull request by Chenditang**

### CODE REVIEW

**Comment by Chenditang** [comment]
The test result on libvirt-1.2.13 fc20.x86_64

\# ./run -k --no-downloads -t libvirt --tests virsh.metadata
SETUP: PASS (0.20 s)
DATA DIR: /var/lib/virt_test
DEBUG LOG: /root/autotest/client/tests/virt/logs/run-2015-07-28-10.55.54/debug.log
TESTS: 9
(1/9) type_specific.io-github-autotest-libvirt.virsh.metadata.positive_test.config_option: PASS (24.46 s)
(2/9) type_specific.io-github-autotest-libvirt.virsh.metadata.positive_test.current_option: PASS (5.56 s)
(3/9) type_specific.io-github-autotest-libvirt.virsh.metadata.positive_test.live_option: PASS (5.55 s)
(4/9) type_specific.io-github-autotest-libvirt.virsh.metadata.positive_test.live_config_option: PASS (24.49 s)
(5/9) type_specific.io-github-autotest-libvirt.virsh.metadata.positive_test.edit_option: PASS (6.37 s)
(6/9) type_specific.io-github-autotest-libvirt.virsh.metadata.negative_test.config_current_option: PASS (13.06 s)
(7/9) type_specific.io-github-autotest-libvirt.virsh.metadata.negative_test.live_current_option: PASS (5.37 s)
(8/9) type_specific.io-github-autotest-libvirt.virsh.metadata.negative_test.invalid_option: PASS (5.36 s)
(9/9) type_specific.io-github-autotest-libvirt.virsh.metadata.negative_test.nonexist_uri: PASS (5.32 s)
TOTAL TIME: 95.58 s (01:35)
TESTS PASSED: 9
TESTS FAILED: 0
SUCCESS RATE: 100.00 %

File: libvirt/tests/src/virsh_cmd/domain/virsh_metadata.py [hunk 1]
```diff
@@ -101,20 +101,22 @@ def get_metadata():
                                         new_metadata=metadata_value,
                                         **virsh_dargs)
                 check_result(result, status_error)
-            if "--config" in metadata_option:
+        # Get metadata
+        for option in metadata_option.split():
```
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/domain/virsh_metadata.py]: check '--config' in string metadata_option is working fine, no need use split and strict check equal.

Comment (Chenditang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_metadata.py]: @waynesun09 
We want to test "--live --config" options.  set the XML metadata with "--live --config" is fine,
but get the XML metadata with "--live --config" will return error, so we need to split metadata_option and then check the result.

\# virsh metadata --domain vm1 --uri http://herp.derp/ --live --config --key herp --set '\<derp xmlns:foobar="http://foo.bar/">foo<bar\>\</bar\>\</derp\>'
Metadata modified

\# virsh metadata --domain vm1 --uri http://herp.derp/ --live --config --key herp
error: flags 'affect live' and 'affect config' in virDomainGetMetadata are mutually exclusive

\# virsh metadata --domain vm1 --uri http://herp.derp/ --live --key herp
\<derp xmlns:foobar="http://foo.bar/">foo<bar\>\</bar\>\</derp\>

\# virsh metadata --domain vm1 --uri http://herp.derp/ --config --key herp
\<derp xmlns:foobar="http://foo.bar/">foo<bar\>\</bar\>\</derp\>

**Comment by will-Do** [comment]
ACK

File: libvirt/tests/src/virsh_cmd/domain/virsh_metadata.py (hunk 1 above)
Comment (waynesun09) [inline libvirt/tests/src/virsh_cmd/domain/virsh_metadata.py]: @Chenditang got it, you are right

**Comment by waynesun09** [comment]
ACK and merging now.

**Pull request merged**